### PR TITLE
Compute SKU support and VM, VMSS integration with 'User Assigned MSI'

### PR DESCRIPTION
**Pull request by anuchandy**
PR includes:

> 1. Compute SKU Support [commit](https://github.com/Azure/azure-libraries-for-net/pull/136/commits/3cd834b90fd99dd2cd967d62cfd6985d0ddba8ac)
> 2. VM integration with User Assigned MSI [commit](https://github.com/Azure/azure-libraries-for-net/pull/136/commits/b0767b4e7cd162bab201cdd75725bf8f7cdfd127)
> 3. VMSS integration with User Assigned MSI [commit](https://github.com/Azure/azure-libraries-for-net/pull/136/commits/70c0d43c3116b0f38bf0f300225c60eb1b1f5246)

**[Compact-Break description]**

With new "User Assigned Managed Service Identity", it is necessary to distinguish it from existing "System Assigned Managed Service Identity". This PR contains the [commit](https://github.com/Azure/azure-libraries-for-net/pull/136/commits/e66a590b0ede61a9d1b65ba0483c7e3a61adf936) that renames existing MSI methods in VM and VMSS to make it clear that those are "System Assigned Managed Service Identities". 

<table>
  <tr>
    <th align=left>Area/Model</th>
    <th align=left>In V1.4</th>
    <th align=left>In V1.6</th>
    <th align=left>Remarks</th>
    <th align=left>Ref</th>
  </tr>
  <tr>
    <td><code>VirtualMachine</code></td>
    <td><code>.ManagedServiceIdentityTenantId()</code></td>
    <td><code>.SystemAssignedManagedServiceIdentityTenantId()</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachine</code></td>
    <td><code>.ManagedServiceIdentityPrincipalId()</code></td>
    <td><code>.SystemAssignedManagedServiceIdentityPrincipalId()</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachine</code></td>
    <td><code>.WithManagedServiceIdentity()</code></td>
    <td><code>.WithSystemAssignedManagedServiceIdentity()</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachine</code></td>
    <td><code>.WithRoleBasedAccessTo(scope, role)</code></td>
    <td><code>.WithSystemAssignedIdentityBasedAccessTo(resourceId, role)</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachine</code></td>
    <td><code>.WithRoleBasedAccessToCurrentResourceGroup(role)</code></td>
    <td><code>.WithSystemAssignedIdentityBasedAccessToCurrentResourceGroup(role)</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachine</code></td>
    <td><code>.WithRoleDefinitionBasedAccessTo(scope, roleDefinitionId)</code></td>
    <td><code>.WithSystemAssignedIdentityBasedAccessTo(resourceId, roleDefinitionId)</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachine</code></td>
    <td><code>.WithRoleDefinitionBasedAccessToCurrentResourceGroup(roleDefinitionId)</code></td>
    <td><code>.WithSystemAssignedIdentityBasedAccessToCurrentResourceGroup(roleDefinitionId)</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachineScaleSet</code></td>
    <td><code>.ManagedServiceIdentityTenantId()</code></td>
    <td><code>.SystemAssignedManagedServiceIdentityTenantId()</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachineScaleSet</code></td>
    <td><code>.ManagedServiceIdentityPrincipalId()</code></td>
    <td><code>.SystemAssignedManagedServiceIdentityPrincipalId()</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachineScaleSet</code></td>
    <td><code>.WithManagedServiceIdentity()</code></td>
    <td><code>.WithSystemAssignedManagedServiceIdentity()</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachineScaleSet</code></td>
    <td><code>.WithRoleBasedAccessTo(scope, role)</code></td>
    <td><code>.WithSystemAssignedIdentityBasedAccessTo(resourceId, role)</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachineScaleSet</code></td>
    <td><code>.WithRoleBasedAccessToCurrentResourceGroup(role)</code></td>
    <td><code>.WithSystemAssignedIdentityBasedAccessToCurrentResourceGroup(role)</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachineScaleSet</code></td>
    <td><code>.WithRoleDefinitionBasedAccessTo(scope, roleDefinitionId)</code></td>
    <td><code>.WithSystemAssignedIdentityBasedAccessTo(resourceId, roleDefinitionId)</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
  <tr>
    <td><code>VirtualMachineScaleSet</code></td>
    <td><code>.WithRoleDefinitionBasedAccessToCurrentResourceGroup(roleDefinitionId)</code></td>
    <td><code>.WithSystemAssignedIdentityBasedAccessToCurrentResourceGroup(roleDefinitionId)</code></td>
    <td></td>
    <td><a href="https://github.com/Azure/azure-libraries-for-net/pull/136">PR #136 </a></td>
  </tr>
</table>

### CODE REVIEW

File: src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent.csproj [hunk 1]
```diff
@@ -58,6 +58,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\Graph.RBAC\Microsoft.Azure.Management.Graph.RBAC.Fluent.csproj" />
+    <ProjectReference Include="..\Msi\Microsoft.Azure.Management.Msi.Fluent.csproj" />
```
Comment (hovsepm) [inline src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent.csproj]: These two needs to be added as package dependencies.

**Comment by hovsepm** [comment]
@anuchandy please pull from upstream and resolve conflicts.

**Comment by hovsepm** [comment]
@anuchandy please fix merge conflicts once more :)

**Comment by anuchandy** [comment]
@hovsepm done

**Pull request merged**